crates/dprint/src/commands/lsp/mod.rs

```diff
@@ -0,0 +1,63 @@
+use tower_lsp::jsonrpc::Result;
+use tower_lsp::lsp_types::DidChangeTextDocumentParams;
+use tower_lsp::lsp_types::DidCloseTextDocumentParams;
+use tower_lsp::lsp_types::DidOpenTextDocumentParams;
+use tower_lsp::lsp_types::DocumentFormattingParams;
+use tower_lsp::lsp_types::DocumentRangeFormattingParams;
+use tower_lsp::lsp_types::InitializeParams;
+use tower_lsp::lsp_types::InitializeResult;
+use tower_lsp::lsp_types::OneOf;
+use tower_lsp::lsp_types::ServerCapabilities;
+use tower_lsp::lsp_types::TextDocumentSyncCapability;
+use tower_lsp::lsp_types::TextDocumentSyncKind;
+use tower_lsp::lsp_types::TextDocumentSyncOptions;
+use tower_lsp::lsp_types::TextEdit;
+use tower_lsp::LanguageServer;
+
+struct Backend;
+
+#[tower_lsp::async_trait]
+impl LanguageServer for Backend {
+  async fn initialize(&self, _: InitializeParams) -> Result<InitializeResult> {
+    Ok(InitializeResult {
+      // TODO: Any server info we want to include here?
```

Comment:
Awesome! Thanks so much for working on this.

I started very briefly setting this up so there's some code you can copy from here https://github.com/dprint/dprint/commit/3b3c4760b412061b802c87c3faeabf3690f1190f You might need to use a channel to get around tower lsp's send + sync requirements.

Reply:
Thanks a lot! I very much welcome contributions and or helpful hints :)

> You might need to use a channel to get around tower lsp's send + sync requirements.

Not sure I understand what you mean :/ Isn't the LSP's connection enough?

Reply:
> Not sure I understand what you mean :/ Isn't the LSP's connection enough?

tower_lsp requires everything to implement the `Send` and `Sync` traits (so it works with a multi-threaded tokio runtime), but dprint uses a single threaded tokio runtime (I opened an issue earlier to drop this requirement in tower-lsp https://github.com/ebkalderon/tower-lsp/issues/386 which would make things more straightforward).

If you try to store any of dprint's structs on `Backend` it probably won't work because they're not `Send` or `Sync`. To get around that you could use a tokio channel (https://docs.rs/tokio/latest/tokio/sync/mpsc/fn.channel.html) to send messages from Backend to a task, process them, then send a response back to `Backend` on a oneshot channel. If you'd like, I can add a basic example to show how that would look.

Reply:
Got it. Thanks a lot for explaining! I would indeed appreciate an example and I can try taking it up from there :)

Reply:
I pushed an example.